src/main/java/io/github/thebusybiscuit/slimefun4/core/services/profiler/SummaryOrderType.java

```diff
@@ -0,0 +1,55 @@
+package io.github.thebusybiscuit.slimefun4.core.services.profiler;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Holds the different types of ordering for summaries.
+ *
+ * @author Walshy
+ */
+public enum SummaryOrderType {
+
+    /**
+     * Sort by highest to the lowest total timings
+     */
+    HIGHEST,
+    /**
+     * Sort by lowest to the highest total timings
+     */
+    LOWEST,
+    /**
+     * Sort by average timings (highest to lowest)
+     */
+    AVERAGE;
+
+    @ParametersAreNonnullByDefault
+    List<Map.Entry<String, Long>> sort(SlimefunProfiler profiler, Set<Map.Entry<String, Long>> entrySet) {
+        switch(this) {
+            case HIGHEST:
+                return entrySet.stream()
+                        .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
```

Comment:
this could also be

`.sorted(Entry.comparingByValue(Comparator.reverseOrder()))`

if you add an import, dont know which one is preferred here